chapters/en/slides/module5/module5_05.Rmd

```diff
@@ -0,0 +1,393 @@
+| chart.encode(alt.Color('Horsepower')))
+ .resolve_scale(color='independent'))
+```
+
+Notes:
+In the plots on this slide,
+we have encoded the same variable (Horsepower) to be represented by both the x-axis and the color.
+This is rarely done in practice,
+but it is useful to contrast and understand color schemes
+because we know that the colors of the points should change in a perceptually uniform manner
+as we move from left to right along the x-axis.
+
+In the leftmost plot in this slide we used the default categorical color scheme.
+As you can see this is not at all well suited to represent numerical data
+as the sharp transitions between hues makes it appear as if there are groups in the data,
+whereas we know that these values are changing gradually from left to right on the x-axis.
+
+In the plot to the right,
+we have used the default sequential color scheme for quantitative data in Altair.
+This conveys the gradual change in the Horsepower values in a perceptually uniform manner.
+We can clearly see that color is changing gradually from light to dark along the x-axis,
+although it can be hard to pinpoint exactly which number a particular color is
+(remember that color is not as good as position to convey an exact value
+even when it is easy to see the overall trend).
+
+---
+
+## Combining lightness variation with a slight variation in hue highlights detail in the data
+
+```{python}
+# TODO hide code
+((chart.encode(alt.Color('Horsepower'))
+| chart.encode(alt.Color('Horsepower', scale=alt.Scale(scheme='viridis', reverse=True))))
+ .resolve_scale(color='independent'))
+```
+
+Notes:
+The leftmost plot here uses the same default color scheme as in the previous slide.
+
+The plot on the right has a color scheme that adds a slight variation in hue
+while keeping the perceptually uniform variation in lightness.
+
+The reason for this is to highlight small variations in the data
+when this is of great importance. 
+We can see the effect of this buy studying the values in the 60-80 range in both plots.
+It is hard to see the differences between these in the leftmost plot,
+but thanks to the hue transition from yellow to green in the rightmost plot
+we can more clearly perceive detail in this range.
+
+Although the hue shifts in this plots are more gradual
+than in the plot with the categorical color scheme in the previous slide,
+it may still introduce a slight bias towards seeing groupings where there are none,
+which is something to keep in mind.
+
+In practice,
+both of these color schemes are used often to represent quantitative data.
+
+---
+
+## High values should have the most contrast to the background
+
+```{python}
+import altair as alt
+from vega_datasets import data
+
+cars = data.cars()
+alt.Chart(cars).mark_point(size=50, filled=True).encode(
+    x='Horsepower',
+    y='Miles_per_Gallon',
+    color=alt.Color('Horsepower', scale=alt.Scale(scheme='viridis', reverse=True)))
```

Comment:
I am guessing that the streange colorlegend issue here comes from vegawidget when knitting (maybe referencing the previus chart somehow). Will revisist when changing to HTML output

![image](https://user-images.githubusercontent.com/4560057/108776504-596dd580-7517-11eb-8bb5-3e5a285df341.png)

Reply:
yup fixed by html